napari/layers/points/points.py

```diff
@@ -1207,7 +1207,8 @@ def selected_data(self, selected_data):
             with self.block_update_properties():
                 self.current_face_color = face_color
 
-        size = list({self.size[i, self._dims_displayed].mean() for i in index})
+        size_index = np.ix_(index, self._dims_displayed)
+        size = np.unique(np.mean(self.size[size_index], axis=0))
```

Comment:
this is super elegant - I wasn't familiar with `np.ix_` and had to step through it carefully and read [the docs](https://numpy.org/doc/stable/reference/generated/numpy.ix_.html), maybe a quick comment explaining what's going on would be useful?

Reply:
Thanks for the comment, as after adding an explanatory comment, I think this is not equivalent to the existing code, as I should be using `axis=1` instead to compute the mean over the displayed dimensions as in `_view_size`.

As a result, I'm also adding a failing test to ensure we captured the desired behavior here. The desired behavior is a little complex (as I understand it) because `size` can vary per-point and per-dimension, but we only use the mean for visualization purposes (at least when not doing out-of-plane slicing where size is also used).

Reply:
Test added and fix made.